nav2_mppi_controller/src/trajectory_visualizer.cpp

```diff
@@ -81,6 +87,8 @@ void TrajectoryVisualizer::add(
   for (size_t i = 0; i < size; i++) {
     add_marker(i);
```

Comment:
The `for` loop is here, you can add in `populate_optimal_path` with `i` so we don't iterate twice